xwiki-platform-core/xwiki-platform-rest/xwiki-platform-rest-server/src/main/java/org/xwiki/rest/internal/resources/classes/StaticListClassPropertyValuesProvider.java

```diff
@@ -49,13 +51,64 @@
 
     @Override
     protected PropertyValues getAllowedValues(StaticListClass propertyDefinition, int limit, String filter)
-        throws Exception
     {
         final PropertyValues result = new PropertyValues();
         List<String> allValues = StaticListClass.getListFromString(propertyDefinition.getValues());
+        Map<String, ListItem> valueMap = StaticListClass.getMapFromString(propertyDefinition.getValues());
+
+        String lowerFilter = filter.toLowerCase();
+
+        allValues.stream().map(id -> constructPropertyValueForId(id, propertyDefinition, valueMap))
+            // Filter both by key and by (translated) label (that's why we construct a PropertyValue first)
+            .filter(val -> val.getValue().toString().toLowerCase().contains(lowerFilter)
+                || (val.getMetaData().containsKey(META_DATA_LABEL)
+                && val.getMetaData().get(META_DATA_LABEL).toString().toLowerCase().contains(lowerFilter)))
+            .limit(limit).forEach((result::withPropertyValues));
 
-        allValues.stream().filter(s -> s.toLowerCase().contains(filter.toLowerCase())).limit(limit)
-            .forEach(item -> result.withPropertyValues(new PropertyValue(item)));
         return result;
     }
+
+    /**
+     * {@inheritDoc}
```

Comment:
From what I see there's actually no doc in `AbstractClassPropertyValuesProvider` for it, so maybe it would worth it to add some doc there. Moreover since you override the implem it might also worth it to explain why you override it here.

Reply:
Good catch, I've added a comment.